package pe.edu.upeu.pharmamobile.di

import io.ktor.client.engine.HttpClientEngine
import io.ktor.client.engine.okhttp.OkHttp
import org.koin.android.ext.koin.androidContext
import org.koin.core.module.Module
import org.koin.dsl.module
import pe.edu.upeu.pharmamobile.domain.platform.Compartidor
import pe.edu.upeu.pharmamobile.platform.CompartidorAndroid

actual val platformModule: Module = module {
    single<HttpClientEngine> {
        OkHttp.create()
    }

    single<Compartidor> {
        CompartidorAndroid(androidContext())
    }
}

actual val backendBaseUrl: String =
    "http://10.0.2.2:8080/"